brd-ios/breadwallet/src/Redesign/Core/Configurable/Library/Controls/ButtonConfigurable.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  ButtonConfigurable.swift
+//  
+//
+//  Created by Rok Cresnik on 28/09/2021.
+//
+
+import Foundation
+
+protocol BaseButtonConfigurable: Configurable {
+    associatedtype BackgroundConfig: BackgorundConfigurable
+    
+    var backgroundConfiguration: BackgroundConfig { get }
+    var selectedBackgroundConfiguration: BackgroundConfig { get }
+    var disabledBackgroundConfiguration: BackgroundConfig { get }
+    
+    func configuration(for state: DisplayState) -> BackgroundConfig
+}
+
+extension BaseButtonConfigurable {
+    func configuration(for state: DisplayState) -> BackgroundConfig {
+        switch state {
+        case .normal:
+            return backgroundConfiguration
+            
+        case .selected,
+                .highlighted:
```

Comment:
indentation issues

Reply:
this is actually not an issue.. its just how multi cases get formatted :S